frontend/src/routes/train.jsx

```diff
@@ -0,0 +1,151 @@
+import { createEffect, createSignal } from "solid-js";
+import { createStore } from "solid-js/store";
+import { styled } from "solid-styled-components";
+import Button from "../components/form/button";
+import InputEmail from "../components/form/email-input";
+import InputHash from "../components/form/hash-input";
+import Slider from "../components/slider";
+import InputFile from "../components/form/file-input";
+import { getFilesUrls } from "../helpers/getFilesUrl";
+import { validateEmail } from "../helpers/validation";
+
+const Train = () => {
+  // store of data from inputs of form
+  const [formData, setFormData] = createStore({
+    email: "",
+    files: [],
+  });
+
+  // store of errors inputs of form
+  const [formErrors, setFormErrors] = createStore({
+    email: "",
+    files: "",
+  });
+
+  const [hash, setHash] = createSignal("");
+
+  const [imagesList, setImagesList] = createSignal([]);
+
+  // set form data when inputs changed
+  const handleFormChange = (event) => {
+    const { target } = event;
+    setFormData(() => {
+      if (target.type === `file`) {
+        return { [target.name]: target.files };
+      } else {
+        return { [target.name]: target.value };
+      }
+    });
+  };
+
+  createEffect(() => {
+    const isValidEmail = validateEmail(formData.email);
+    const isValidFiles = formData.files.length > 10;
+
+    if (isValidEmail) {
+      setFormErrors({ email: "" });
```

Comment:
The same as [here](https://github.com/ShkarupaDC/3d-scene-editing-app/pull/1/files#r1133153367)